### ISSUE TITLE
Optionally warn about unsaved changes

### ISSUE THREAD

**Issue by anewuser** [pipes-digital/pipes#116]
Some users might appreciate a warning before leaving the editor page with unsaved changes, as in this example:

![uBlock Origin](https://user-images.githubusercontent.com/6733770/183311423-fdd02bc4-61ab-424b-b0cf-d560d25a1a64.png)

If you decide to implement this, though, please make it optional. Personally, I'd disable it, since I'm used to opening different pipes just to check or copy something. I consciously close editor tabs and expect this to undo any accidental changes.

**Comment by onli** [comment]
Hm. That it should be disabled makes me think it might not be the right solution for the data loss problem.

Wouldn't this be better: Save the temporary changes in the browser and restore them when going back to the canves, **iff** it is exactly the same pipe? Probably should be combined with a marker for which changes are not saved yet (something in the block title?).

**Comment by anewuser** [comment]
Yes, your idea is less intrusive. What about also adding an option to revert all changes that have been done locally to the canvas since the last time the pipe was saved? If you like this idea, maybe you can also move the `save` and `Pipe Output` links to the top, since they aren't visible when all block categories are expanded (on resolutions equal or smaller than 1920×1080): 

![Pipes](https://user-images.githubusercontent.com/6733770/190902359-7f79e877-2763-42bd-8901-49007636900e.png)
